### PR TITLE
[.NET] Return warning when validated inputs don't have an error message

### LINKED ISSUE #4449: [.NET][Input Validation] Return warning if validation properties exist without an errorMessage
# Platform

What platform is your issue or question related to? (Delete other platforms).

* .NET WPF

# Details

If the card author has set input validation properties but not set an errorMessage, return a warning to encourage the card author to add an errorMessage to provide more information to the user explaining why the input is invalid.

### PR DESCRIPTION
**Pull request by RebeccaAnne**
## Related Issue
Fixes #4449

## Description
Add warnings for validated inputs without error messages for .NET.

## How Verified
.NET tests passed. Confirmed expected behavior using existing [MissingErrorMessages.json](https://github.com/microsoft/AdaptiveCards/blob/main/samples/v1.3/Tests/MissingErrorMessages.json) sample card.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/4885)

### CODE REVIEW

File: source/dotnet/Library/AdaptiveCards.Rendering.Wpf.Xceed/XceedDateInput.cs [hunk 1]
```diff
@@ -40,6 +40,12 @@ public static FrameworkElement Render(AdaptiveDateInput input, AdaptiveRenderCon
                 datePicker.Style = context.GetStyle("Adaptive.Input.Date");
                 datePicker.DataContext = input;
 
+                if (input.IsRequired && input.ErrorMessage == null)
+                {
+                    context.Warnings.Add(new AdaptiveWarning((int)AdaptiveWarning.WarningStatusCode.NoWarningForValidatedInput,
+                        "Inputs with validation should include an ErrorMessage"));
```
Comment (shalinijoshi19) [inline source/dotnet/Library/AdaptiveCards.Rendering.Wpf.Xceed/XceedDateInput.cs:46]: @RebeccaAnne do we have a plan to localize these error strings that we bubble out via the AdaptiveWarning/Error object(s)?

File: source/dotnet/Library/AdaptiveCards.Rendering.Wpf.Xceed/XceedTimeInput.cs [hunk 2]
```diff
@@ -41,6 +41,12 @@ public static FrameworkElement Render(AdaptiveTimeInput input, AdaptiveRenderCon
                 timePicker.Style = context.GetStyle("Adaptive.Input.Time");
                 timePicker.DataContext = input;
 
+                if (input.IsRequired && input.ErrorMessage == null)
+                {
+                    context.Warnings.Add(new AdaptiveWarning((int)AdaptiveWarning.WarningStatusCode.NoWarningForValidatedInput,
+                        "Inputs with validation should include an ErrorMessage"));
```
Comment (shalinijoshi19) [inline source/dotnet/Library/AdaptiveCards.Rendering.Wpf.Xceed/XceedTimeInput.cs:47]: Is there a common place in the shared .NET library that we could house this string as a variable to reference in all these places vs needing to hardcode the literal each place?

File: source/dotnet/Library/AdaptiveCards.Rendering.Wpf.Xceed/XceedDateInput.cs [hunk 3]
```diff
@@ -40,6 +40,12 @@ public static FrameworkElement Render(AdaptiveDateInput input, AdaptiveRenderCon
                 datePicker.Style = context.GetStyle("Adaptive.Input.Date");
                 datePicker.DataContext = input;
 
+                if (input.IsRequired && input.ErrorMessage == null)
```
Comment (shalinijoshi19) [inline source/dotnet/Library/AdaptiveCards.Rendering.Wpf.Xceed/XceedDateInput.cs:43]: What;s our behavior for empty or whitespace-only values for the ErrorMessage?

Comment (RebeccaAnne) [reply source/dotnet/Library/AdaptiveCards.Rendering.Wpf.Xceed/XceedDateInput.cs]: Good point. Fixed.

---
In reply to: [501254812](https://github.com/microsoft/AdaptiveCards/pull/4885#discussion_r501254812) [](ancestors = 501254812)

File: source/dotnet/Library/AdaptiveCards.Rendering.Wpf.Xceed/XceedDateInput.cs (hunk 1 above)
Comment (RebeccaAnne) [reply source/dotnet/Library/AdaptiveCards.Rendering.Wpf.Xceed/XceedDateInput.cs:46]: There's not currently any plan to localize warning strings. I added it to the doc i'm working on that covers possible areas for localization.

---
In reply to: [501253546](https://github.com/microsoft/AdaptiveCards/pull/4885#discussion_r501253546) [](ancestors = 501253546)

**Review by shalinijoshi19 — APPROVED** [review]

**Pull request merged**